src/query/storage/cache/redis_cache.go

```diff
@@ -203,28 +192,28 @@ func (cache *RedisCache) Get(entries []*storage.FetchQuery) []*storage.PromResul
 func (cache *RedisCache) Set(
 	entries []*storage.FetchQuery,
 	values []*storage.PromResult,
+	prefix string,
 ) error {
-	var commands []radix.CmdAction
+	var args []string
 	// Place holder variable to fit radix.Cmd() functions
 	var response string
 
 	for i := range entries {
-		// Skip over non-existent results
-		if entries[i] == nil {
-			continue
-		}
-		entry := keyEncode(entries[i])
+		entry := KeyEncode(entries[i], prefix)
+		// Encode PromResult into string for Redis to store
 		value, err := resultEncode(values[i])
 		cache.logger.Info("Prom result size", zap.Int("length", len(value)), zap.Int("ts_len", len(values[i].PromResult.Timeseries)))
 		if err != nil {
 			cache.logger.Error("Redis encode error", zap.Error(err))
 			continue
 		}
 
-		commands = append(commands, radix.Cmd(&response, "SET", entry, value), radix.Cmd(&response, "EXPIRE", entry, ExpirationTime))
+		args = append(args, entry, value)
+		// commands = append(commands, radix.Cmd(&response, "SET", entry, value), radix.Cmd(&response, "EXPIRE", entry, ExpirationTime))
 	}
 
-	if err := cache.client.Do(radix.Pipeline(commands...)); err != nil {
+	// Use MSET and don't set expiration, let Redis LRU determine the process
+	if err := cache.client.Do(radix.Cmd(&response, "MSET", args...)); err != nil {
```

Comment:
Do we figure out a metric from Redis that we can check the really memory usage in the last X hours.

Reply:
I think the MEMORY STATS command from Redis has some metrics that we can record to calculate that, will look into it further